Pets_script.sql

```diff
@@ -0,0 +1,109 @@
+
+SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
+SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
+SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';
+
+-- -----------------------------------------------------
+-- Schema mydb
+-- -----------------------------------------------------
+DROP SCHEMA IF EXISTS `mydb` ;
+
+-- -----------------------------------------------------
+-- Schema mydb
+-- -----------------------------------------------------
+CREATE SCHEMA IF NOT EXISTS `mydb` DEFAULT CHARACTER SET utf8 ;
+USE `mydb` ;
+
+-- -----------------------------------------------------
+-- Table `mydb`.`persons`
+-- -----------------------------------------------------
+DROP TABLE IF EXISTS `mydb`.`persons` ;
+
+CREATE TABLE IF NOT EXISTS `mydb`.`persons` (
+  `idpersons` INT NOT NULL,
+  `name` VARCHAR(45) NOT NULL,
+  `email` VARCHAR(60) NOT NULL,
+  `password` VARCHAR(45) NOT NULL,
+  PRIMARY KEY (`idpersons`))
+ENGINE = InnoDB;
+
+CREATE UNIQUE INDEX `idpersons_UNIQUE` ON `mydb`.`persons` (`idpersons` ASC) VISIBLE;
+
+CREATE UNIQUE INDEX `email_UNIQUE` ON `mydb`.`persons` (`email` ASC) VISIBLE;
+
+
+-- -----------------------------------------------------
+-- Table `mydb`.`pets`
+-- -----------------------------------------------------
+DROP TABLE IF EXISTS `mydb`.`pets` ;
+
+CREATE TABLE IF NOT EXISTS `mydb`.`pets` (
+  `idpets` INT NOT NULL,
+  `name` VARCHAR(45) NOT NULL,
+  `persons_idpersons` INT NOT NULL,
+  PRIMARY KEY (`idpets`, `persons_idpersons`),
+  CONSTRAINT `fk_pets_persons1`
+    FOREIGN KEY (`persons_idpersons`)
+    REFERENCES `mydb`.`persons` (`idpersons`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
```

Comment:
Please take look at other `ON DELETE` options, one of them may be more appropriate here

Reply:
I think "ON DELETE SET NULL" is better for pets table,  "ON DELETE CASCADE" is better for schedule table and "ON DELETE CASCADE" is better for schedule_has_pets table.